internal/pkg/consensus/processor.go

```diff
@@ -645,3 +676,53 @@ func (p *DefaultProcessor) minerOwnerAddress(ctx context.Context, st state.Tree,
 	}
 	return address.NewFromBytes(ret[0])
 }
+
+func getOrCreateActor(ctx context.Context, st *state.CachedTree, store vm.StorageMap, addr address.Address, gt *vm.GasTracker) (*actor.Actor, address.Address, error) {
+	// resolve address before lookup
+	idAddr, found, err := ResolveAddress(ctx, addr, st, store, gt)
+	if err != nil {
+		return nil, address.Undef, err
+	}
+
+	if found {
+		act, err := st.GetActor(ctx, idAddr)
+		return act, idAddr, err
+	}
+
+	if addr.Protocol() != address.ID {
+		initAct, err := st.GetActor(ctx, address.InitAddress)
+		if err != nil {
+			return nil, address.Undef, err
+		}
+
+		// this should never fail due to lack of gas since gas doesn't have meaning here
+		noopGT := vm.NewGasTracker()
+		noopGT.MsgGasLimit = 10000
```

Comment:
nit: ok for now but this magic number carries lots of questions with it for someone like me who hasn't fully internalized gas processing rules.  An explanation pre-emptively answering some of these (is this always enough, is this some special protocol defined number, what about the gas price?) would be helpful.